src/main/scala/org/hammerlab/guacamole/commands/VAFHistogram.scala

```diff
@@ -182,14 +181,15 @@ object VAFHistogram {
         sc.parallelize(histogramOutput).saveAsTextFile(args.output)
       } else {
         // Print histograms to standard out
-        variantAlleleHistograms.foreach(histogram =>
+        for {
+          (_, histogram) <- variantAlleleHistograms
+        } {
           histogram.toSeq.sortBy(_._1).foreach(kv => println(histogramToString(kv)))
-        )
+        }
       }
 
       if (args.cluster) {
-        val numClusters = args.numClusters
-        variantLoci.foreach(buildMixtureModel(_, numClusters))
+        buildMixtureModel(variantLoci, args.numClusters)
```

Comment:
yea, that's right and basically intentional, though I should have explained my thinking!

I view this as two changes in sequence:

1. spark apps that are pushing multiple datasets (samples) through a pipeline where each is independent of the others, like this was doing (e.g. `.map` over an array of `RDD`s) should be changed to only process one sample at a time, and the calling scripts / workflows should handle that multiplexing over multiple samples instead. In my experience, pushing that [sequential processing of multiple RDDs] into one Spark app has disadvantages / Spark is the wrong part of the stack to do that. https://github.com/bigdatagenomics/adam/issues/833 has some context where I wanted to move ADAM cmds toward performing multiple independent "commands" within a given Spark app, but was talked out of it and convinced that the opposite approach is superior. Spark doesn't parallelize its processing of an array of `RDD`s, so it's really the worst possible place to multiplex on samples, since no parallelization is possible.

2. given the above, and imagining this command now only operates on a single sample, I've generalized it to be able to handle multiple samples in one pass / as part of one mixture model; this is different than the previous "multiple independent samples/models in one app" setup and does actually let Spark operate on multiple samples at once, albeit to answer a different question than was being answered before, but per 1. the previous question is better answered by parallelizing elsewhere in the stack.

Sorry for the wall of text, lmk if that doesn't make sense.